misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/DataMapperNodeVisitor.java

```diff
@@ -0,0 +1,118 @@
+import io.ballerina.compiler.api.symbols.VariableSymbol;
+import io.ballerina.compiler.syntax.tree.MappingConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.ballerina.compiler.syntax.tree.SpreadFieldNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.VariableDeclarationNode;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Visitor to extract Record Type Structure information.
+ */
+public class DataMapperNodeVisitor extends NodeVisitor {
+    public final HashMap<String, String> restFields;
+    public final HashMap<String, Map<String, RecordFieldSymbol>> spreadFields;
+    public final ArrayList<String> specificFieldList;
+    private String rhsSymbolName;
+    private SemanticModel model;
+
+    public DataMapperNodeVisitor(String rhsSymbolName) {
+        this.restFields = new HashMap<>();
+        this.spreadFields = new HashMap<>();
+        this.rhsSymbolName = rhsSymbolName;
+        this.specificFieldList = new ArrayList<>();
+    }
+
+    public void setModel(SemanticModel model) {
+        this.model = model;
+    }
+
+    @Override
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
+        if (variableDeclarationNode.initializer().isPresent()) {
+            if (variableDeclarationNode.initializer().get().kind() == SyntaxKind.MAPPING_CONSTRUCTOR) {
+                String rightSymbolName = variableDeclarationNode.typedBindingPattern().typeDescriptor().toString().
+                        replaceAll(" ", "");
+                rightSymbolName = rightSymbolName.replaceAll("\n", "");
+                // to check if the variable if from another module
+                if (rightSymbolName.contains(":")) {
+                    String[] rightSymbolNameArray = rightSymbolName.split(":");
+                    rightSymbolName = rightSymbolNameArray[rightSymbolNameArray.length - 1];
+                }
+                if (this.rhsSymbolName.equals(rightSymbolName)) {
+                    MappingConstructorExpressionNode mappingConstructorExpressionNode =
+                            (MappingConstructorExpressionNode) variableDeclarationNode.initializer().get();
+                    SeparatedNodeList<MappingFieldNode> fields = mappingConstructorExpressionNode.fields();
+                    for (MappingFieldNode field : fields) {
+                        if (field.kind() == SyntaxKind.SPECIFIC_FIELD) {
+                            SpecificFieldNode field1 = (SpecificFieldNode) field;
+                            String fieldName = field1.fieldName().toString().replaceAll(" ", "");
+                            if (field1.valueExpr().isPresent() && fieldName.contains("\"")) {
```

Comment:
Can the field name contain `"`?

Reply:
When the field is a rest field, there are `"` in the field name.